holoviews/operation/downsampling.py

```diff
@@ -0,0 +1,168 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import math
+
+import numpy as np
+import param
+
+from .resampling import ResamplingOperation1D
+
+def _argmax_area(prev_x, prev_y, avg_next_x, avg_next_y, x_bucket, y_bucket):
+    """Vectorized triangular area argmax computation.
+    Parameters
+    ----------
+    prev_x : float
+        The previous selected point is x value.
+    prev_y : float
+        The previous selected point its y value.
+    avg_next_x : float
+        The x mean of the next bucket
+    avg_next_y : float
+        The y mean of the next bucket
+    x_bucket : np.ndarray
+        All x values in the bucket
+    y_bucket : np.ndarray
+        All y values in the bucket
+    Returns
+    -------
+    int
+        The index of the point with the largest triangular area.
+    """
+    return np.abs(
+        x_bucket * (prev_y - avg_next_y)
+        + y_bucket * (avg_next_x - prev_x)
+        + (prev_x * avg_next_y - avg_next_x * prev_y)
+    ).argmax()
+
+
+def _lttb_inner(x, y, n_out, sampled_x, offset):
+    a = 0
+    for i in range(n_out - 3):
+        o0, o1, o2 = offset[i], offset[i + 1], offset[i + 2]
+        a = (
+            _argmax_area(
+                x[a],
+                y[a],
+                np.mean(x[o1:o2]),
+                y[o1:o2].mean(),
+                x[o0:o1],
+                y[o0:o1],
+            )
+            + offset[i]
+        )
+        sampled_x[i + 1] = a
+
+    # ------------ EDGE CASE ------------
+    # next-average of last bucket = last point
+    sampled_x[-2] = (
+        _argmax_area(
+            x[a],
+            y[a],
+            x[-1],  # last point
+            y[-1],
+            x[offset[-2] : offset[-1]],
+            y[offset[-2] : offset[-1]],
+        )
+        + offset[-2]
+    )
+
+
+def _lttb(x, y, n_out):
+    """
+    Downsample the data using the LTTB algorithm (python implementation).
+
+    Args:
+        x (np.ndarray): The x-values of the data.
+        y (np.ndarray): The y-values of the data.
+        n_out (int): The number of output points.
+    Returns:
+        np.array: The indexes of the selected datapoints.
+    """
+    # Bucket size. Leave room for start and end data points
+    block_size = (y.shape[0] - 2) / (n_out - 2)
+    # Note this 'astype' cast must take place after array creation (and not with the
+    # aranage() its dtype argument) or it will cast the `block_size` step to an int
+    # before the arange array creation
+    offset = np.arange(start=1, stop=y.shape[0], step=block_size).astype(np.int64)
+
+    # Construct the output array
+    sampled_x = np.empty(n_out, dtype="int64")
```

Comment:
These dtypes should be unsigned integers. A case could be made to reduce them to 32 bits.